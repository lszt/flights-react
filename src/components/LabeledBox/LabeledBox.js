import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  margin: 0 0 2em 0;
  border: 1px solid #eee;
  box-shadow: 3px;
`;

const Label = styled.div`
  border-bottom: 1px solid ${props => props.theme.colors.main};
  padding: 1em;
  background-color: ${props => props.theme.colors.background};
`;

const Content = styled.div`
  padding: ${props => typeof props.padding === 'number' ? `${props.padding}px` : '1em'};
  ${props => props.contentMaxHeight > 0 && `max-height: ${props.contentMaxHeight}px;`}
  
  a {
    text-decoration: underline;
  }

  p:not(:last-child) {
    margin-bottom: 1em;
  }
`;

const LabeledBox = props => (
  <Wrapper className={props.className} innerRef={props.innerRef}>
    <Label>{props.label}</Label>
    <Content
      maxHeight={props.contentMaxHeight}
      padding={props.contentPadding}
    >{props.children}</Content>
  </Wrapper>
);

LabeledBox.propTypes = {
  label: PropTypes.string.isRequired,
  className: PropTypes.string,
  contentMaxHeight: PropTypes.number,
  contentPadding: PropTypes.number,
  innerRef: PropTypes.func,
};

export default LabeledBox;
