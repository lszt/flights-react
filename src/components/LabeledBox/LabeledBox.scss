.LabeledBox {
  margin: 0 0 2em 0;
  overflow: hidden;
  border: 1px solid #eee;
  box-shadow: 3px;

  > .label {
    border-bottom: 1px solid #ed351c;
    padding: 1em;
    background-color: #f6f6f6;
  }

  > .content {
    padding: 1em;
    overflow: hidden;
  }
}
