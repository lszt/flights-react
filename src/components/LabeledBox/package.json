{
  "name": "LabeledBox",
  "version": "0.0.0",
  "private": true,
  "main": "./LabeledBox.js"
}
