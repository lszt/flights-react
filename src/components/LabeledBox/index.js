import LabeledBox from './LabeledBox';

export default LabeledBox;
