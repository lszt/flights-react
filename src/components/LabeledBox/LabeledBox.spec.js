import React from 'react';
import {shallow} from 'enzyme';
import LabeledBox from './LabeledBox';

describe('components', () => {
  describe('LabeledBox', () => {
    it('is built correctly', () => {
      const labeledBox = (
        <LabeledBox label="My label" className="my-class">
          <div>child1</div>
          <div>child2</div>
        </LabeledBox>
      );

      const component = shallow(labeledBox);

      expect(component.debug()).toMatchSnapshot();
    });
  });
});
