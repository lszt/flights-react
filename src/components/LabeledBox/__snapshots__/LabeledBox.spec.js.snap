// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components LabeledBox is built correctly 1`] = `
"<styled.div className=\\"my-class\\" innerRef={[undefined]}>
  <styled.div>
    My label
  </styled.div>
  <styled.div maxHeight={[undefined]} padding={[undefined]}>
    <div>
      child1
    </div>
    <div>
      child2
    </div>
  </styled.div>
</styled.div>"
`;
